pkgs/development/libraries/icu/make-icu.nix

```diff
@@ -8,9 +8,13 @@ let
 
   pname = "icu4c";
 
+  release = if lib.hasSuffix "rc" version
+            then "${lib.replaceStrings [ "1" ] [ "" ] (lib.replaceStrings [ "." ] [ "-" ] version)}"
+            else "${lib.replaceStrings [ "." ] [ "-" ] version}";
```

Comment:
```suggestion
  release = lib.replaceStrings [ "1" ] [ "" ]
    (if lib.hasSuffix "rc" version then (lib.replaceStrings [ "." ] [ "-" ] version) else lib.id);
```

Reply:
Thanks, @SuperSandro2000, my intent was to keep this PR open until the final release is published and then remove the commit that allows to build rc, unless of course this seems useful for nixpkgs.
What are your thoughts on this?

Reply:
I think the question is what is the motivation to open the PR based on an RC. To get ofBorg builds? That sounds like a repeatable motivation, so maybe the -rc handling changes can be merged if you are interested in opening such RC-based PRs in the future.

Reply:
Correct, @7c6f434c, given that icu is such a "low-level" package and updating the default icu pckage is likely going to cause a mass-rebuild I wanted to get a head start on the upcoming version.

I'll have a look at previous rc releases to see whether the diverge in the same manner from the official releases.

Reply:
Except for the tags `icu4j-cldr-1-9-rc` and `release-64-2-rc` tags for icu release candidates do not contain a "minor" version (see below).
If folks see value in adding support for icu rc releases to the nixpkgs I'm happy to clean up this PR.

What are your thoughts on this @7c6f434c and @SuperSandro2000?

<details><summary>icu rc tags</summary>

```sh
curl https://api.github.com/repos/unicode-org/icu/git/refs/tags \
  | jq .[].ref \
  | sed -e 's@refs/tags/@@' \
  | sort -V \
  | ag 'rc$'
icu4j-cldr-1-9-rc
icu4j-release-49-rc
icu4j-release-50-rc
icu4j-release-51-rc
icu4j-release-52-rc
icu4j-release-53-rc
icu4j-release-54-rc
icu4j-release-55-rc
icu4j-release-56-rc
icu4j-release-57-rc
icu4j-release-58-rc
icu-release-49-rc
icu-release-50-rc
icu-release-51-rc
icu-release-52-rc
icu-release-53-rc
icu-release-54-rc
icu-release-55-rc
icu-release-56-rc
icu-release-57-rc
icu-release-58-rc
release-59-rc
release-60-rc
release-61-rc
release-62-rc
release-63-rc
release-64-2-rc
release-64-rc
release-65-rc
release-66-rc
release-67-rc
release-68-rc
release-69-rc
release-70-rc
release-71-rc
release-72-rc
release-73-rc
release-74-rc
release-75-rc
release-76-rc
```


</details>